File: totosuki/ABC/B/307.py
N = int(input())
S = [input() for _ in range(N)]
for i in range(N):
  for j in range(N):
    if i == j:
      continue
    t = S[i] + S[j]
    t_r = "".join(list(reversed(t)))
    if t == t_r:
      print("Yes")
      exit()

print("No")

File: OCH_Tao/ABC/B/307.js
const [N, ...S] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
for (let i = 0; i < parseInt(N); i++) {
  for (let j = 0; j < parseInt(N); j++) {
    if (i === j) {
      continue;
    }
    const X = S[i] + S[j];
    if (X === [...X].reverse().join("")) {
      console.log("Yes");
      process.exit();
    }
  }
}
console.log("No");